invmul_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/crate-crypto/go-ipa/bandersnatch/fp"
+	"github.com/stretchr/testify/require"
+	blst "github.com/supranational/blst/bindings/go"
+)
+
+func BenchmarkFpInverse(b *testing.B) {
+	b.Run("go-ipa", func(b *testing.B) {
+		var x fp.Element
+		x.SetRandom()
+
+		b.ReportAllocs()
+		b.ResetTimer()
+		var resIPA fp.Element
+		for i := 0; i < b.N; i++ {
+			resIPA.Inverse(&x)
+		}
+	})
+	b.Run("go-blst", func(b *testing.B) {
+		var xIPA fp.Element
+		xIPA.SetRandom()
+		var xBLST blst.Scalar
+		beBytes := xIPA.Bytes()
+		xBLST.FromBEndian(beBytes[:])
+
+		var resBLST *blst.Scalar
+		b.ReportAllocs()
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			resBLST = xBLST.Inverse()
+		}
+		_ = resBLST
+	})
+}
+
+func TestVerifyGoIPAvsBLSTInverse(t *testing.T) {
+	t.Parallel()
+
+	for i := 0; i < 500_000; i++ {
+		// Generate random field element
+		var xIPA fp.Element
+		xIPA.SetRandom()
+
+		// Get the inverse with go-ipa
+		var invIPA fp.Element
+		invIPA.Inverse(&xIPA)
+
+		// Transform the same x to blst, and calculate the multiplication.
+		var xBLST blst.Scalar
+		beBytes := xIPA.Bytes()
+		xBLST.FromBEndian(beBytes[:])
+		invBLST := xBLST.Inverse()
+
+		// Both inverses should match.
+		invIPABytes := invIPA.Bytes()
+		require.Equal(t, invIPABytes[:], invBLST.Serialize())
+	}
+}
```

Comment:
I created this test to be sure that both `go-ipa` and `blst` agree on results for the inverse for the same scalar.